praw/models/reddit/live.py

```diff
@@ -281,6 +281,55 @@ def close(self):
         url = API_PATH['live_close'].format(id=self.thread.id)
         self.thread._reddit.post(url)
 
+    def update(self, **settings):
+        """Update settings of the live thread.
+
+        The following settings are supported:
+
+        :param title: (Optional) The title of the live thread.
+        :param description: (Optional) The live thread's description.
+        :param nsfw: (Optional) Indicate whether this thread is not safe for
+            work.
+        :param resources: (Optional) Markdown formatted information that is
+            useful for the live thread.
+        :returns: If any settings are given, return new :class:`.LiveThread`
+            instance (updated settings can be retrieved through the instance.)
+            If no settings are given, raise ``TypeError``.
+
+        Unspecified settings will maintain their current value.
+
+        Usage:
+
+        .. code-block:: python
+
+           thread = reddit.live('xyu8kmjvfrww')
+           new_title = '[nsfw]' + thread.title
+           old_description = thread.description
+           updated_thread = thread.contrib.update(title=new_title, nsfw=True)
+           assert updated_thread.title == new_title
+           assert updated_thread.description == old_description
+
+        """
+        available_settings = ('title', 'description', 'nsfw', 'resources')
+        settings = {key: settings[key] for key in available_settings
+                    if key in settings}
+
+        if not settings:
+            raise TypeError('No settings are given.')
+        elif len(settings) == len(available_settings):
+            new_settings = settings
+        else:
+            thread = LiveThread(self.thread._reddit, self.thread.id)
```

Comment:
Why not use `self.thread`? I suspect maybe you want to get the latest version of said attributes as they could be stale on `self.thread`. However, that's still a possibility even after this fetch so I'm not sure it's worth the potential extra request.